### PR TITLE
Task/lattice 2973 linker memory

### PR DESCRIPTION
**Pull request by njhlai**
`SocratesMatcher` currently initialise with a copy of the model every time a new thread is started. These (copies of the) model actually live off heap according to dl4j’s documentation, and thus are not managed by the JVM’s gc. This can cause memory issues. 

Instead, wrap the model with `ParallelInference` and let dl4j manage the multi-threading.

EDIT: ```LINKER_OFFHEAP``` should be no more than half of ```LINKER_XMX```. Based on what I see on prod before, maybe something like ```LINKER_XMX=-Xmx50g```, ```LINKER_OFFHEAP=-Dorg.bytedeco.javacpp.maxbytes=20g``` should suffice.

### CODE REVIEW

**Comment by njhlai** [comment]
In production, `LINKER_OFFHEAP` needs to be defined, and `LINKER_XMX` and `LINKER_OFFHEAP` needs to be tuned as well.

File: linker/src/main/kotlin/com/openlattice/linking/matching/SocratesMatcher.kt [hunk 1]
```diff
@@ -50,10 +52,14 @@ class SocratesMatcher(
         private val linkingFeedbackService: PostgresLinkingFeedbackService
 ) : Matcher {
 
-    private var localModel = ThreadLocal.withInitial { model.clone() }
+    private val localModel = ParallelInference.Builder(model)
+                        .inferenceMode(InferenceMode.SEQUENTIAL)
+                        // .batchLimit(32)
```
Comment (njhlai) [inline linker/src/main/kotlin/com/openlattice/linking/matching/SocratesMatcher.kt:57]: Need decision whether to do this sequentially or in batches.

Comment (geekbeast) [reply linker/src/main/kotlin/com/openlattice/linking/matching/SocratesMatcher.kt:57]: That doesn't sound very parallel.

Comment (njhlai) [reply linker/src/main/kotlin/com/openlattice/linking/matching/SocratesMatcher.kt:57]: Batching makes the most sense here I think, just want to see if anyone has strong opinions/if I miss anything.

**Review by geekbeast — APPROVED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Pull request merged**